### ISSUE TITLE
update fields issue in reduxForm HOC

### ISSUE THREAD

**Issue by jsdmc** [erikras/redux-form#507]
*Scenario*
I have a form with some field initialized with value 0. This field is binded to text input. When I try to remove value - actual field value becomes "", but in text input it still 0.

*Issue*
Here https://github.com/erikras/redux-form/blob/master/src/createHigherOrderComponent.js#L45 in second comparison we have a problem because by default 'deep-equal' module uses non strict comparison and thus we skip updating for fields values (but in state they are updated)
```javascript```
componentWillReceiveProps(nextProps) {
  if (!deepEqual(this.props.fields, nextProps.fields) || !deepEqual(this.props.form, nextProps.form)) {
    this.fields = readFields(nextProps, this.props, this.fields, this.asyncValidate, isReactNative);
  }
  ...
}
```

Solution is quite easy, we just need to pass parameters for comparison
```javascript```
deepEqual(this.props.form, nextProps.form, { strict: true })
```

**Comment by jsdmc** [comment]
PR https://github.com/erikras/redux-form/pull/508

**Comment by erikras** [comment]
Fix published in [`v4.1.1`](https://github.com/erikras/redux-form/releases/tag/v4.1.1).
